test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +215,55 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
-void is_about(float x, float y)
+template <typename>
+struct get_eps;
+
+template <>
+struct get_eps<float>
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
+    static constexpr float val = 1e-6;
+};
 
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
+template <>
+struct get_eps<double>
+{
+    static constexpr double val = 1e-11;
+};
 
-CLANG_DIAGNOSTIC_POP
-}
+template <>
+struct get_eps<long double>
+{
+    static constexpr long double val = 1e-11;
+};
 
-void is_about(double x, double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(T x, T y)
```

Comment:
I would suggest to add tolerance as the last function parameter, with default value tol = get_eps<T>::val
It makes is_about function more useful.